diff -urN paho.mqtt.c-1.3.9/src/CMakeLists.txt paho.mqtt.c-1.3.9.bak/src/CMakeLists.txt
--- paho.mqtt.c-1.3.9/src/CMakeLists.txt	2021-09-24 17:46:07.000000000 +0800
+++ paho.mqtt.c-1.3.9.bak/src/CMakeLists.txt	2021-09-24 17:43:28.000000000 +0800
@@ -60,7 +60,8 @@
     SET(LIBS_SYSTEM ws2_32 crypt32 RpcRT4)
 ELSEIF (UNIX)
     IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
-        SET(LIBS_SYSTEM c dl pthread rt)
+        # SET(LIBS_SYSTEM c dl pthread rt)
+        SET(LIBS_SYSTEM c dl z)
         # anl is only available with glibc so check if it is found before using
         # it or build will fail on uclibc or musl
         FIND_LIBRARY(LIB_ANL anl)
@@ -106,8 +107,8 @@
     
     SET_TARGET_PROPERTIES(
         paho-mqtt3c paho-mqtt3a PROPERTIES
-        VERSION ${CLIENT_VERSION}
-        SOVERSION ${PAHO_VERSION_MAJOR}
+        # VERSION ${CLIENT_VERSION}
+        # SOVERSION ${PAHO_VERSION_MAJOR}
         COMPILE_DEFINITIONS "PAHO_MQTT_EXPORTS=1")
 
     IF(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
@@ -157,8 +158,8 @@
     ENDIF()       
     SET_TARGET_PROPERTIES(
         paho-mqtt3c-static paho-mqtt3a-static PROPERTIES
-        VERSION ${CLIENT_VERSION}
-        SOVERSION ${PAHO_VERSION_MAJOR}
+        # VERSION ${CLIENT_VERSION}
+        # SOVERSION ${PAHO_VERSION_MAJOR}
         COMPILE_DEFINITIONS "PAHO_MQTT_STATIC=1")
 
     FOREACH(TARGET paho-mqtt3c-static paho-mqtt3a-static)
@@ -202,8 +203,8 @@
     
         SET_TARGET_PROPERTIES(
             paho-mqtt3cs paho-mqtt3as PROPERTIES
-            VERSION ${CLIENT_VERSION}
-            SOVERSION ${PAHO_VERSION_MAJOR}
+            # VERSION ${CLIENT_VERSION}
+            # SOVERSION ${PAHO_VERSION_MAJOR}
             COMPILE_DEFINITIONS "OPENSSL=1;PAHO_MQTT_EXPORTS=1")
 
         IF(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
@@ -254,8 +255,8 @@
 
         SET_TARGET_PROPERTIES(
             paho-mqtt3cs-static paho-mqtt3as-static PROPERTIES
-            VERSION ${CLIENT_VERSION}
-            SOVERSION ${PAHO_VERSION_MAJOR}
+            # VERSION ${CLIENT_VERSION}
+            # SOVERSION ${PAHO_VERSION_MAJOR}
             COMPILE_DEFINITIONS "OPENSSL=1;PAHO_MQTT_STATIC=1")
         IF (NOT WIN32)
             SET_TARGET_PROPERTIES(paho-mqtt3cs-static PROPERTIES OUTPUT_NAME paho-mqtt3cs)
