diff -urN cJSON-1.7.15_org/CMakeLists.txt cJSON-1.7.15/CMakeLists.txt
--- cJSON-1.7.15_org/CMakeLists.txt	2021-08-25 19:15:09.000000000 +0800
+++ cJSON-1.7.15/CMakeLists.txt	2021-09-22 18:52:07.094214664 +0800
@@ -163,10 +163,10 @@
     install(EXPORT "${CJSON_LIB}" DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/cJSON")
 endif()
 
-set_target_properties("${CJSON_LIB}"
-    PROPERTIES
-        SOVERSION "${CJSON_VERSION_SO}"
-        VERSION "${PROJECT_VERSION}")
+# set_target_properties("${CJSON_LIB}"
+#     PROPERTIES
+#         SOVERSION "${CJSON_VERSION_SO}"
+#         VERSION "${PROJECT_VERSION}")
 
 #cJSON_Utils
 option(ENABLE_CJSON_UTILS "Enable building the cJSON_Utils library." OFF)
